### DISCUSSION TITLE
Schema intersection and new parser functionality.

### DISCUSSION THREAD

**Discussion by NTDLS** [NTDLS/Katzebase#92] [Announcements]
Merged the final part of the two major rewrites. I do not expect any further large breaking changes.

Lots of fixes, speed improvements, UI issues resolved, static analyzer fixes and updates.
The vast majority of the changes are to StaticSchemaIntersectionProcessor, which was literally rewritten from the ground up. This may introduce bugs, but it MMUCH more readable, understandable, optimizable, performant and maintainable.

oh, and Scaler -> Scalar 😲 

## Added support for updating with join:
![image](https://github.com/user-attachments/assets/64a67d07-1ae1-4c7b-83e3-472ce31c92f9)

### So finally you can query and order by ```count(*) desc```

![image](https://github.com/user-attachments/assets/a62ca627-7582-41d2-956c-bad923f3066c)

![image](https://github.com/user-attachments/assets/a8e9c0fc-6cbf-4c9f-8466-e221879f8fd7)

## Can now connect to more than one server:
![image](https://github.com/user-attachments/assets/6e2cff55-effb-4ddc-905b-e59824de587e)

## Code collapse improvements:
![image](https://github.com/user-attachments/assets/1ecf91e6-1f9b-4dc0-b361-f336e5e65e3f)

## Schema awareness:
![image](https://github.com/user-attachments/assets/978cc53c-504b-4112-b569-28f2b734e16f)

**Comment by NTDLS** [comment]
@ingted Check out the notes on [0.22.0](https://github.com/NTDLS/Katzebase/releases/tag/0.22.0) and [0.23.0](https://github.com/NTDLS/Katzebase/releases/tag/0.23.0).

Also, check out [Regression.kbs](https://github.com/NTDLS/Katzebase/blob/master/%40Scripts/Regression.kbs). This is a major milestone! The script has no parser issues and executes successfully. If you run it in the UI, it probably wont show the results grids due to to many result-sets. Regardless, it works!!

I also think I got all the tests you made working with the new project structure.

**Comment by ingted** [comment]
Cool!! I will read the code of StaticSchemaIntersectionProcessor carefully!!
